packages/NodeRemoval/NodeRemover.php

```diff
@@ -43,12 +43,12 @@ public function removeNodeFromStatements(
         Class_ | ClassMethod | Function_ $nodeWithStatements,
         Node $toBeRemovedNode
     ): void {
-        foreach ((array) $nodeWithStatements->stmts as $key => $stmt) {
+        foreach ((array) $nodeWithStatements->stmts as $stmt) {
             if ($toBeRemovedNode !== $stmt) {
                 continue;
             }
 
-            unset($nodeWithStatements->stmts[$key]);
+            $this->removeNode($stmt);
```

Comment:
Here the `unset()` should be kept, as removing node immediatlly and locally is more clear.
The `removeNode()` method puts the responsibility in hidden logic, that I wanted to refactor away.

Especially in case of params/args/stmts.

Reply:
Ok, I updated to keep `unset()` usage https://github.com/rectorphp/rector-src/pull/3255/commits/aace4e2caa9a193b157197b607fe096b4375ce26